packages/lodestar/src/chain/factory/block/index.ts

```diff
@@ -23,26 +23,26 @@ export async function assembleBlock(
   randaoReveal: Bytes96,
   graffiti = ZERO_HASH
 ): Promise<BeaconBlock> {
-  const [parentBlock, stateContext] = await Promise.all([chain.getHeadBlock(), chain.getHeadStateContext()]);
-  const parentHeader: BeaconBlockHeader = blockToHeader(config, parentBlock!.message);
-  const headState = stateContext!.state.clone();
+  const head = chain.forkChoice.getHead();
+  const stateContext = await chain.regen.getBlockSlotState(head.blockRoot, slot - 1);
```

Comment:
I don't think -1 should be here, it says in spec that state should be rolled to assigned slot not assigned slot -1:

> Note: In this section, state is the state of the slot for the block proposal without the block yet applied. That is, state is the previous_state processed through any empty slots up to the assigned slot using process_slots(previous_state, slot).

Reply:
lmao, wow, we were never doing this right...